R/classicalpredictionperformance.R

```diff
@@ -0,0 +1,654 @@
+        tempFunnelPlot$plotObject <- tempPlot
+      }
+
+    }
+  }
+
+  return()
+}
+.metamiscFitFunnelAsymmetryNames <- function(shortcut) {
+  switch(
+    shortcut,
+    "E-UW"  = "Egger (unweighted)",
+    "E-FIV" = "Egger (multiplicative overdispersion)",
+    "M-FIV" = "Macaskill",
+    "M-FPV" = "Macaskill (pooled)",
+    "P-FPV" = "Peters",
+    "D-FIV" = "Debray"
+  )
+}
+.metamiscFitFunnelAsymmetryOptions <- function(shortcut) {
+  switch(
+    shortcut,
+    "E-UW"  = "funnelAsymmetryTestEggerUW",
+    "E-FIV" = "funnelAsymmetryTestEggerFIV",
+    "M-FIV" = "funnelAsymmetryTestMacaskillFIV",
+    "M-FPV" = "funnelAsymmetryTestMacaskillFPV",
+    "P-FPV" = "funnelAsymmetryTestPeters",
+    "D-FIV" = "funnelAsymmetryTestDebrayFIV"
+  )
+}
+.metamiscFitFunnelAsymmetryggPlot  <- function(x, ref, xlab = "Effect size",
+                                               confint = TRUE, confint.level = 0.1, confint.alpha = .50, confint.col = "skyblue") {
+
+  if (!inherits(x, "fat"))
+    stop("Argument 'x' must be an object of class \"fat\".")
+  if (confint.level < 0 | confint.level > 1) {
+    stop("Argument 'confint.level' must be between 0 and 1.")
+  }
+
+  xval <- x$model$data[, "y"]
+  if (x$method %in% c("E-UW", "E-FIV")) {
+    ylab <- "Standard error"
+    yval <- (x$model$data[, "x"])
+    ylim <- rev(c(0, max(yval, na.rm = TRUE)))
+    xlim <- c(min(c(0, xval)), max(xval))
+  } else if (x$method %in% c("M-FIV")) {
+    ylab <- "Sample size"
+    yval <- (x$model$data[, "x"])
+    ylim <- (c(0, max(yval, na.rm = TRUE)))
+    xlim <- c(min(c(0, xval)), max(xval))
+  } else if (x$method == "P-FPV") {
+    ylab <- "Sample size"
+    yval <- (x$model$data[, "x"])
+    ylim <- rev(c(0, max(yval, na.rm = TRUE)))
+    xlim <- c(min(c(0, xval)), max(xval))
+    step <- ((max(yval) - min(yval))/5)
+    yax  <- c(plyr::round_any(1/min(yval), 10^(sapply(round(1/min(yval)), nchar) - 1)),
+              plyr::round_any(1/seq(step, 4 * step, by = step), 10), plyr::round_any(1/max(yval), 10))
+  } else if (x$method == "D-FIV") {
+    ylab <- "Total events"
+    yval <- (x$model$data[, "x"])
+    ylim <- rev(c(0, max(yval, na.rm = TRUE)))
+    xlim <- c(min(c(0, xval)), max(xval))
+    step <- ((max(yval) - min(yval))/4)
+    yax  <- c(plyr::round_any(1/min(yval), 10^(sapply(round(1/min(yval)), nchar) - 1)),
+              plyr::round_any(1/seq(step, 4 * step, by = step), 10), plyr::round_any(1/max(yval), 10))
+  } else if (x$method == "D-FAV") {
+    ylab <- "Total events"
+    yval <- (x$model$data[, "x"])
+    ylim <- rev(c(0, max(yval, na.rm = TRUE)))
+    xlim <- c(min(c(0, xval)), max(xval))
+    step <- ((max(yval) - min(yval))/4)
+    yax  <- c(plyr::round_any(1/min(yval),10^(sapply(round(1/min(yval)), nchar) - 1)),
+              plyr::round_any(1/seq(step, 4 * step, by = step), 10), plyr::round_any(1/max(yval), 10))
+  } else {
+    stop("Plot not supported!")
```

Comment:
```suggestion
    stop("Plot not supported!", domain = NA)
```
otherwise, people will actually translate this.